sql/parse/parse.go

```diff
@@ -1293,6 +1294,31 @@ func convertMultiAlterDDL(ctx *sql.Context, query string, c *sqlparser.MultiAlte
 			return nil, err
 		}
 	}
+
+	// certain alter statements need to happen before others
+	sort.Slice(statements, func(i, j int) bool {
```

Comment:
is there any way to test this better? the tests added above are mostly just error checks, would be nice to confirm that either the positive case works as expected and/or this doesn't have untested edge cases. Even just unit tests that just make sure the reordering works.